lib/ansible/modules/windows/setup.ps1

```diff
@@ -318,11 +318,12 @@ if($gather_subset.Contains('platform')) {
 
     $ansible_facts += @{
         ansible_architecture = $win32_os.OSArchitecture
-        ansible_domain = $ip_props.DomainName
-        ansible_fqdn = ($ip_props.Hostname + "." + $ip_props.DomainName)
-        ansible_hostname = $env:COMPUTERNAME
+        ansible_domain = $win32_cs.Domain.Substring($win32_cs.Workgroup.length)
+        ansible_fqdn = ($win32_cs.DNSHostname + "." + $win32_cs.Domain.Substring($win32_cs.Workgroup.length))
+        ansible_hostname = $win32_cs.DNSHostname
+        ansible_netbios_name = $win32_cs.Name
         ansible_kernel = $osversion.Version.ToString()
-        ansible_nodename = ($ip_props.HostName + "." + $ip_props.DomainName)
+        ansible_nodename = ($win32_cs.DNSHostname + "." + $win32_cs.Domain.Substring($win32_cs.Workgroup.length))
```

Comment:
If we are reusing values like before it could be beneficial to store that in a var and just set the var here instead of calculating it twice.

Reply:
Changed